gnovm/tests/files/zrealm_crossrealm11.gno

```diff
@@ -0,0 +1,152 @@
+// PKGPATH: gno.land/r/crossrealm_test
+package crossrealm_test
+
+import (
+	"std"
+
+	ptests "gno.land/p/demo/tests"
+	"gno.land/p/demo/ufmt"
+	rtests "gno.land/r/demo/tests"
+	testfoo "gno.land/r/demo/tests_foo"
+)
+
+func getRealmCaller() std.Address {
+	return std.GetRealmCaller()
+}
+
+func Exec(fn func()) {
+	fn()
+}
+
+func main() {
+	// Create a map of the potential callers, this will give more understandable
+	// output than the bech32 addresses.
+	callersByAddr := make(map[std.Address]string)
+	for _, caller := range []string{
+		"user1.gno", "gno.land/r/crossrealm_test", "gno.land/r/demo/tests",
+	} {
+		addr := std.DerivePkgAddr(caller)
+		callersByAddr[addr] = caller
+	}
+
+	baseCallStack := "user1.gno -> r/crossrealm_test.main()"
+	tests := []struct {
+		callStackAdd string
+		callerFn     func() std.Address
+	}{
+		{
+			callStackAdd: "",
+			callerFn:     std.GetRealmCaller,
+		},
+		{
+			callStackAdd: " -> r/crossrealm_test.getRealmCaller()",
+			callerFn:     getRealmCaller,
+		},
+		{
+			callStackAdd: " -> p/demo/tests",
+			callerFn:     ptests.GetRealmCaller,
+		},
+		{
+			callStackAdd: " -> p/demo/tests -> p/demo/tests/subtests",
+			callerFn:     ptests.GetSubtestsCaller,
+		},
+		{
+			callStackAdd: " -> r/demo/tests",
+			callerFn:     rtests.GetRealmCaller,
+		},
+		{
+			callStackAdd: " -> r/demo/tests -> r/demo/tests/subtests",
+			callerFn:     rtests.GetSubtestsCaller,
+		},
+		{
+			callStackAdd: " -> p/demo/tests -> r/demo/tests",
+			callerFn:     ptests.GetRealmCaller,
+		},
```

Comment:
I think we lack a last test: `r > p`.

Reply:
And maybe also `r > p > r` to be complete.

Reply:
I see now. The third test handles it.

Apologies for the initial confusion. Your test is exhaustive, although it took some effort to understand at first. Looks good to me.